aptos-move/framework/aptos-framework/sources/reconfiguration.move

```diff
@@ -151,6 +154,87 @@ module aptos_framework::reconfiguration {
+        borrow_global<SlowReconfigureStatus>(@aptos_framework).deadline
+    }
+
+    public(friend) fun start_slow_reconfigure(account: &signer) acquires SlowReconfigureStatus {
+        assert!(!slow_reconfigure_in_progress(), error::invalid_state(EANOTHER_RECONFIGURATION_IN_PROGRESS));
+        let status = borrow_global_mut<SlowReconfigureStatus>(@aptos_framework);
+        status.in_progress = true;
+        status.deadline = timestamp::now_seconds() + 999999999;
+        stake::disable_validator_set_changes();
+        std::config_for_next_epoch::disable_updates(account);
+        //TODO: start DKG.
+    }
+
+    public(friend) fun update_slow_reconfigure(params: vector<u8>) acquires SlowReconfigureStatus {
+        assert!(slow_reconfigure_in_progress(), error::invalid_state(ENO_RECONFIGURATION_IN_PROGRESS));
+        std::debug::print(&params) //TODO: consume `params`
+    }
+
+    public(friend) fun terminate_slow_reconfigure(account: &signer) acquires Configuration, SlowReconfigureStatus {
+        assert!(slow_reconfigure_in_progress(), error::invalid_state(ENO_RECONFIGURATION_IN_PROGRESS));
+        move_from<SlowReconfigureStatus>(@aptos_framework);
+
+        // Basically do what `reconfigure()` does. But here we need the signer.
+
+        if (chain_status::is_genesis() || timestamp::now_microseconds() == 0 || !reconfiguration_enabled()) {
+            return
+        };
+
+        let config_ref = borrow_global_mut<Configuration>(@aptos_framework);
+        let current_time = timestamp::now_microseconds();
+
+        // Reconfiguration "forces the block" to end, as mentioned above. Therefore, we must process the collected fees
+        // explicitly so that staking can distribute them.
+        //
+        // This also handles the case when a validator is removed due to the governance proposal. In particular, removing
+        // the validator causes a reconfiguration. We explicitly process fees, i.e. we drain aggregatable coin and populate
+        // the fees table, prior to calling `on_new_epoch()`. That call, in turn, distributes transaction fees for all active
+        // and pending_inactive validators, which include any validator that is to be removed.
+        if (features::collect_and_distribute_gas_fees()) {
+            // All transactions after reconfiguration are Retry. Therefore, when the next
+            // block starts and tries to assign/burn collected fees it will be just 0 and
+            // nothing will be assigned.
+            transaction_fee::process_collected_fees();
+        };
+
+        stake::on_new_epoch();
+        storage_gas::on_reconfig();
+        consensus_config::on_new_epoch(account);
+        execution_config::on_new_epoch(account);
+        gas_schedule::on_new_epoch(account);
+        std::version::on_new_epoch(account);
+        features::on_new_epoch(account);
+        // TODO: complete the list.
```

Comment:
I think the list should contain on-chain states that may be accessed in rust directly, and here are what I have identified. Anything I missed?

@zekun000 @movekevin